ga4gh/versioning.py

```diff
@@ -0,0 +1,152 @@
+        only can be accessed in development mode
+    active - live and stable; able to be accessed in any mode
+    deactivated - not able to be accessed in any mode
+
+Enabled and disabled versions:
+    these two versions specify the min and max versions at which the
+    endpoints can be accessed
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+
+
+class Lifecycle(object):
+    """
+    A lifecycle stage for a route group
+    """
+    # TODO more lifecycle stages?
+    # having a deprecated status doesn't really make any sense if there's
+    # no programatic difference from ACTIVE (for instance, an endpointStatus
+    # field in the response json indicating deprecation)
+    DEVELOPMENT = "DEVELOPMENT"
+    ACTIVE = "ACTIVE"
+    DEACTIVATED = "DEACTIVATED"
+
+
+class Version(object):
+    """
+    A major/minor/revision version tag
+    """
+    @classmethod
+    def parseString(cls, versionString):
+        versions = versionString.strip('vV').split('.')
+        return Version(*versions)
+
+    def __init__(self, major, minor, revision):
+        self.major = major
+        self.minor = minor
+        self.revision = revision
+
+    def __cmp__(self, other):
+        majorCmp = cmp(self.major, other.major)
+        if majorCmp != 0:
+            return majorCmp
+        minorCmp = cmp(self.minor, other.minor)
+        if minorCmp != 0:
+            return minorCmp
+        else:
+            return cmp(self.revision, other.revision)
+
+    def __hash__(self):
+        return hash((self.major, self.minor, self.revision))
+
+    def __eq__(self, other):
+        return (self.major == other.major and
+                self.minor == other.minor and
+                self.revision == other.revision)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+class RouteGroup(object):
+    """
+    A group that a route belongs to
+    """
+    def __init__(self, lifecycleStage=Lifecycle.DEVELOPMENT,
+                 enabledVersion=None, disabledVersion=None):
+        self.lifecycleStage = lifecycleStage
+        if enabledVersion is not None:
+            self.enabledVersion = Version.parseString(enabledVersion)
+        else:
+            self.enabledVersion = None
+        if disabledVersion is not None:
+            self.disabledVersion = Version.parseString(disabledVersion)
+        else:
+            self.disabledVersion = None
+
+
+class VersioningManager(object):
+
+    # TODO externalize data fields to yaml file?
+    supportedVersionsData = ['0.5.1']
+
+    groupData = {
+        'V_0.5.1_RELEASE': {
+            'enabledVersion': '0.5.1',
+        },
+    }
+
+    def __init__(self, app):
+        self.groups = self._initGroups()
+        self.supportedVersions = self._initSupportedVersions()
+        self.app = app
+
+    def _initSupportedVersions(self):
```

Comment:
This makes sense if we're going to externalize the data as proposed in line 95, but otherwise couldn't we just declare ```supportedVersionsData = set([Version.parseString('0.5.1')])```?

And does it even make sense to externalize the data? If we require code changes for new versions, we might as well make part of the process being updating this code. If we don't require code changes for new versions (other than a generated protocol schema update), it might make sense to externalize, but then we'll need a process for managing multiple schema definitions?